django/db/models/sql/compiler.py

```diff
@@ -523,6 +523,28 @@ def as_sql(self, with_limits=True, with_col_aliases=False):
             if for_update_part and not self.connection.features.for_update_after_from:
                 result.append(for_update_part)
 
+            if self.query.subquery and extra_select:
+                # If the query is used as a subquery the extra selects would
+                # result in more columns than the left hand side expression is
+                # expecting. This can happen when a subquery uses a combination
+                # of order_by() and distinct(); forcing the ordering expressions
+                # to be selected as well. Wrap the query into another subquery
+                # to exclude extraneous selects.
+                subselects = []
```

Comment:
My instinct would be to use a `sub_` prefix for these names (possibly confirmed by an existing sub_params name in other code).